### PR TITLE
Create Hello World app for android

### PR DESCRIPTION
**Pull request by ravitb**
The app does not have any UI, only logics parts.
    The app shows the following functionality:
    1. Create Kin  client
    2. Create Kin account that represent the app user
    3. On board the user to Kin wallet
    4. Send Kins from the user to a different wallet
    6. Gets the user's balance

### CODE REVIEW

File: hello-world/build.gradle [hunk 1]
```diff
@@ -0,0 +1,41 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 28
+
+
+
+    defaultConfig {
+        applicationId "com.example.hello_world"
```
Comment (yosriz) [inline hello-world/build.gradle]: change package and application id to `kin.sdk.hello_world` for consistency

File: hello-world/build.gradle [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+
+    implementation 'com.android.support:appcompat-v7:28.0.0'
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    testImplementation 'junit:junit:4.12'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    implementation 'com.github.kinecosystem:kin-sdk-android:one_blockhain_changes-SNAPSHOT' // TODO: make sure that this line is updated
```
Comment (yosriz) [inline hello-world/build.gradle]: move to top, remove todo

File: hello-world/src/androidTest/java/com/example/hello_world/ExampleInstrumentedTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.example.hello_world;
```
Comment (yosriz) [inline hello-world/src/androidTest/java/com/example/hello_world/ExampleInstrumentedTest.java]: delete example test

File: hello-world/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.hello_world">
```
Comment (yosriz) [inline hello-world/src/main/AndroidManifest.xml]: change package and application id to kin.sdk.hello_world for consistency

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
+package com.example.hello_world;
```
Comment (yosriz) [inline hello-world/src/main/java/com/example/hello_world/MainActivity.java]: change package and application id to kin.sdk.hello_world for consistency

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        // Kin account the is the entity that holds Kins
+        kinAccount = getKinAccount(APP_INDEX);
+
+        // listener for balance changes
+        addBalanceListeners(kinAccount);
+
+        // async function to get kin balance
+        getKinBalance(kinAccount);
+
+        // create the wallet with the Kin account keypair
```
Comment (yosriz) [inline hello-world/src/main/java/com/example/hello_world/MainActivity.java]: I think worth noting that is created on the blockchain and also get funded

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        // as it is an async request, at the callback we will be able to transfer Kins and check the account balance
+        onBoardAccount(kinAccount, new Callbacks() {
+            @Override
+            public void onSuccess() {
+                Log.e(TAG, "Onboarding succeeded");
```
Comment (yosriz) [inline hello-world/src/main/java/com/example/hello_world/MainActivity.java]: Log.d (not an error)

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+                    public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                        fireOnFailure(callbacks, e);
+                    }
+
+                    @Override
+                    public void onResponse(@NonNull Call call, @NonNull Response response) {
+                        int code = response.code();
+                        response.close();
+                        if (code != 200) {
+                            fireOnFailure(callbacks,
```
Comment (yosriz) [inline hello-world/src/main/java/com/example/hello_world/MainActivity.java]: why do you need to fire the callbacks back to the main method?
you can print the success failure here for simplicity.

File: hello-world/src/test/java/com/example/hello_world/ExampleUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
+package com.example.hello_world;
```
Comment (yosriz) [inline hello-world/src/test/java/com/example/hello_world/ExampleUnitTest.java]: remove example test

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+            @Override
+            public void onError(Exception e) {
+                e.printStackTrace();
+            }
+
+        });
+    }
+
+    public void addBalanceListeners(KinAccount account) {
+        account.addBalanceListener(
```
Comment (yosriz) [inline hello-world/src/main/java/com/example/hello_world/MainActivity.java]: I think it worth adding remove listeners example (on activity `onStart` `onDestory`)

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+
+        okHttpClient = new OkHttpClient.Builder()
+            .connectTimeout(20, TimeUnit.SECONDS)
+            .readTimeout(20, TimeUnit.SECONDS)
+            .build();
+
+        // Kin client is the accounts manager for Kin accounts
+        kinClient = new KinClient(this, Environment.TEST, STUB_APP_ID, HELLO_WORLD);
+
+        // Kin account the is the entity that holds Kins
```
Comment (cnotethegr8) [inline hello-world/src/main/java/com/example/hello_world/MainActivity.java]: Kin account is the entity that holds Kins

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        okHttpClient = new OkHttpClient.Builder()
+            .connectTimeout(20, TimeUnit.SECONDS)
+            .readTimeout(20, TimeUnit.SECONDS)
+            .build();
+
+        // Kin client is the accounts manager for Kin accounts
```
Comment (cnotethegr8) [inline hello-world/src/main/java/com/example/hello_world/MainActivity.java]: Kin client is the manager for Kin accounts

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+        setContentView(R.layout.activity_main);
+
+        handler = new Handler(Looper.getMainLooper());
+        okHttpClient = new OkHttpClient.Builder()
+            .connectTimeout(20, TimeUnit.SECONDS)
+            .readTimeout(20, TimeUnit.SECONDS)
+            .build();
+
+        kinClient = new KinClient(this, Environment.TEST, STUB_APP_ID, HELLO_WORLD);
+        kinAccount = getKinAccount(APP_INDEX);
```
[the remark this answers is not in the document]
Comment (ravitb) [reply hello-world/src/main/java/com/example/hello_world/MainActivity.java]: Added a comment.

File: hello-world/build.gradle (hunk 1 above)
Comment (ravitb) [reply hello-world/build.gradle]: It is not part of the sdk

File: hello-world/src/main/java/com/example/hello_world/MainActivity.java [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+    private static final BigDecimal AMOUNT_KIN =  new BigDecimal(SUM_TO_TRANSFER);
+    private OkHttpClient okHttpClient;
+    private Handler handler;
+    private ListenerRegistration listenerRegistration;
+    private KinClient kinClient;
+    private KinAccount kinAccount;
+    private kin.sdk.Request<Transaction> buildTransactionRequest;
+    private kin.sdk.Request<TransactionId> sendTransactionRequest;
+
+    public interface Callbacks {
```
[the remark this answers is not in the document]
Comment (ravitb) [reply hello-world/src/main/java/com/example/hello_world/MainActivity.java]: It helps the visibility of the example. Will be decided later.

**Comment by yosriz** [comment]
@ravitb let's make sure this can be run on travis, and also that there are no issues with jitpack after the changes.

**Pull request closed without merging**